src/app/(post-login)/submission/components/ProgramMenu.tsx

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2023 The Ontario Institute for Cancer Research. All rights reserved
+ *
+ * This program and the accompanying materials are made available under the terms of
+ * the GNU Affero General Public License v3.0. You should have received a copy of the
+ * GNU Affero General Public License along with this program.
+ *  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT
+ * SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+ * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+'use client';
+
+import { css } from '@/lib/emotion';
+import { MenuItem } from '@icgc-argo/uikit';
+import Link from 'next/link';
+import { usePathname } from 'next/navigation';
+import { useState } from 'react';
+
+export default function ProgramMenu({
+	programs,
+	searchQuery,
+}: {
+	programs: { shortName: string }[];
+	searchQuery: string;
+}) {
+	const pathname = usePathname();
+	const [activeProgramIndex, setActiveProgramIndex] = useState(-1);
+
+	const filteredPrograms = programs.filter(
+		({ shortName }) => !searchQuery.length || shortName.search(new RegExp(searchQuery, 'i')) > -1,
+	);
+
+	return (
+		<>
+			<Link
+				href="/submission"
+				css={css`
+					text-decoration: none !important;
+				`}
+			>
+				<MenuItem
+					level={2}
+					content="All Programs"
+					onClick={() => setActiveProgramIndex(-1)}
+					selected={pathname === '/submission'}
+				/>
+			</Link>
+
+			{filteredPrograms.map((program, programIndex) => (
+				<MenuItem
+					level={2}
```

Comment:
🤔 weird non-declarative constant. surely a uikit thing.
possibly a computable value? Not sure what/how to improve here

Reply:
better UIKit would be the improvement